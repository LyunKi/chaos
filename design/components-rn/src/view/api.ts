import { PropsWithChildren } from 'react';
import { GestureResponderEvent, View } from 'react-native';
import { CloudDesignWrap } from '../common';

export interface BasicViewProps {
  onPress?: (event: GestureResponderEvent) => any;
}

export type ViewProps = PropsWithChildren<
  CloudDesignWrap<BasicViewProps, View>
>;
