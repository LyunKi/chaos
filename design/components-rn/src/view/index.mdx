import { Title, Canvas, Meta, Controls, Source } from '@storybook/blocks';

import * as Stories from './index.stories';

<Meta of={Stories} />

<Title />

## Props

<Controls />

## Stories

<Canvas of={Stories.Default} />
