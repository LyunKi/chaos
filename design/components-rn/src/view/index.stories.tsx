import type { Meta, StoryObj } from '@storybook/react';
import React from 'react';
import { Text } from '../index';
import { View } from './index';

const meta = {
  title: 'Cloud-Design/View',
  component: View,
} satisfies Meta<typeof View>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: <Text value="Hello, world!" />,
    ts: {
      width: '$rem:20',
      justifyContent: 'center',
      background: '$color.brand.500',
    },
  },
};
