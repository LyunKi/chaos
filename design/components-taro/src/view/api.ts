import { PropsWithChildren } from 'react';
import { GestureResponderEvent, ViewStyle } from 'react-native';
import { CloudDesignWrap } from '../common';

export interface BasicViewProps {
  onPress?: (event: GestureResponderEvent) => any;
}

export type ViewProps = PropsWithChildren<
  CloudDesignWrap<BasicViewProps, ViewStyle>
>;
