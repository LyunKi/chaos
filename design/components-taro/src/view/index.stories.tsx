import type { Meta, StoryObj } from '@storybook/react';
import React from 'react';
import { Text } from '../index';
import { View } from './index';

const meta = {
  title: 'Cloud-Design/View',
  component: View,
  args: {
    ts: {
      width: '100%',
      height: '$rem:3',
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: '$color.brand.500',
    },
  },
} satisfies Meta<typeof View>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: <Text value="Hello, world!" />,
  },
};
