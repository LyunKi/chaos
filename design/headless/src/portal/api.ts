import React from 'react';

export interface PortalProps {
  container?: HTMLElement;
  children: React.ReactNode;
}
