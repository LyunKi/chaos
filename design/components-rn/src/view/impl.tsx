import React, { forwardRef } from 'react';
import {
  View as RnView,
  ScrollView as RnScrollView,
  StyleSheet,
  TouchableWithoutFeedback,
  Platform,
  GestureResponderEvent,
} from 'react-native';
import { styles } from '@cloud-dragon/common-utils';
import { ThemeManager } from '../common';
import { ViewProps } from './api';

export const View = forwardRef(
  (
    {
      ts,
      style,
      children,
      onPress,
      onLongPress,
      scrollable,
      stopPropagation,
    }: ViewProps,
    ref?: React.Ref<RnView>
  ) => {
    const ViewInstance = scrollable ? RnScrollView : RnView;
    const isInteractive = onPress || onLongPress;
    const Inner = (
      <ViewInstance
        ref={ref}
        style={StyleSheet.flatten([
          { flexDirection: 'row', boxSizing: 'border-box' },
          styles(
            [!!isInteractive, { cursor: 'pointer' }],
            [stopPropagation, { cursor: 'default' }]
          ),
          ThemeManager.themed(ts),
          style,
        ])}
      >
        {children}
      </ViewInstance>
    );
    const realOnPress = (e: GestureResponderEvent) => {
      if (stopPropagation) {
        e.stopPropagation();
      }
      onPress?.(e);
    };
    return isInteractive || stopPropagation ? (
      <TouchableWithoutFeedback onLongPress={onLongPress} onPress={realOnPress}>
        {Inner}
      </TouchableWithoutFeedback>
    ) : (
      Inner
    );
  }
);
