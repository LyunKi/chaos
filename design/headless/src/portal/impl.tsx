import ReactDOM from 'react-dom';
import { PortalProps } from './api';

export const Portal = ({
  container = document.body,
  children,
}: PortalProps) => {
  return ReactDOM.createPortal(children, container);
};
