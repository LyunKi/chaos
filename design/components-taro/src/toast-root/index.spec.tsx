import { render } from '@testing-library/react-native';
import React from 'react';
import { composeStories } from '@storybook/react';
import { PortalProvider } from '@gorhom/portal';
import { View } from '../view';
import * as stories from './index.stories';

const { Default } = composeStories(stories);

describe('ToastRoot', () => {
  it('should match snapshot', () => {
    const rendered = render(
      <View>
        <PortalProvider>
          <Default />
        </PortalProvider>
      </View>
    );
    expect(rendered).toMatchSnapshot();
  });
});
