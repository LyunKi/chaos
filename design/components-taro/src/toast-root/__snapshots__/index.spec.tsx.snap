// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastRoot should match snapshot 1`] = `
<View
  style={
    {
      "boxSizing": "border-box",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      {
        "boxSizing": "border-box",
        "flexDirection": "column",
        "gap": 16,
        "width": 200,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "none",
          "flexDirection": "column",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
          "zIndex": undefined,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderRadius": undefined,
            "boxSizing": "border-box",
            "flexDirection": "row",
            "gap": 8,
            "height": undefined,
            "justifyContent": "center",
            "outline": "unset",
            "paddingHorizontal": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": undefined,
              "fontSize": undefined,
              "fontWeight": undefined,
            }
          }
        >
          simple info
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderRadius": undefined,
            "boxSizing": "border-box",
            "flexDirection": "row",
            "gap": 8,
            "height": undefined,
            "justifyContent": "center",
            "outline": "unset",
            "paddingHorizontal": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": undefined,
              "fontSize": undefined,
              "fontWeight": undefined,
            }
          }
        >
          info
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderRadius": undefined,
            "boxSizing": "border-box",
            "flexDirection": "row",
            "gap": 8,
            "height": undefined,
            "justifyContent": "center",
            "outline": "unset",
            "paddingHorizontal": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": undefined,
              "fontSize": undefined,
              "fontWeight": undefined,
            }
          }
        >
          warning
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderRadius": undefined,
            "boxSizing": "border-box",
            "flexDirection": "row",
            "gap": 8,
            "height": undefined,
            "justifyContent": "center",
            "outline": "unset",
            "paddingHorizontal": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": undefined,
              "fontSize": undefined,
              "fontWeight": undefined,
            }
          }
        >
          error
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderRadius": undefined,
            "boxSizing": "border-box",
            "flexDirection": "row",
            "gap": 8,
            "height": undefined,
            "justifyContent": "center",
            "outline": "unset",
            "paddingHorizontal": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": undefined,
              "fontSize": undefined,
              "fontWeight": undefined,
            }
          }
        >
          success
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderRadius": undefined,
            "boxSizing": "border-box",
            "flexDirection": "row",
            "gap": 8,
            "height": undefined,
            "justifyContent": "center",
            "outline": "unset",
            "paddingHorizontal": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": undefined,
              "fontSize": undefined,
              "fontWeight": undefined,
            }
          }
        >
          loading
        </Text>
      </View>
    </View>
  </View>
</View>
`;
